### DISCUSSION TITLE
Private repo giving an error message

### DISCUSSION THREAD

**Discussion by Rowan-Paul** [upptime/upptime#167] [Q&A] [answered]
When I try to access my site I get redirected to the error page, with console messages saying they can't GET certain paths. The repository is private, when I change it to public (and the normal GitHub API), it just works.

I am using a custom subdomain for my status page and am hosting a GitHub API proxy to send the requests. I have a personal access token configured. The proxy works when I try it in Postman.

When the repo is public but I still use my proxy, the site shows the homepage, but doesn't load the site as it gives an error on ``GET https://github-api-proxy.herokuapp.com/{username}/{repo}/master/history/summary.json``. This endpoint also doesn't seem to work in Postman when using the normal GitHub API.

![image](https://user-images.githubusercontent.com/46132597/103482980-60644b80-4de4-11eb-8150-50a29e963861.png)

The first image is from my normal browser, the second an incognito tab. They seem to give different errors depending on the caching or something, but both don't load the sites.

![image](https://user-images.githubusercontent.com/46132597/103483041-d1a3fe80-4de4-11eb-8df1-8fe7f324f5d6.png)

And it works when the apiBaseUrl is not set and the repo is public:

![image](https://user-images.githubusercontent.com/46132597/103483131-8b02d400-4de5-11eb-809a-a5ce795f0dd1.png)

Here is a copy of [my configuration file](https://gist.github.com/Rowan-Paul/62dc9a4a30abb09aac55520916217184) and a screenshot of the error page with console open:

![image](https://user-images.githubusercontent.com/46132597/103482462-512fce80-4de1-11eb-8988-c5e2a7accead.png)

I can't find out what I'm doing wrong and couldn't find anything online so hoping someone here can help!

**Comment by Rowan-Paul** [comment, marked as answer]
I've fixed it in by creating a server like in this answer from @flauc (thx for the answer): 

 - https://github.com/upptime/upptime/discussions/54#discussioncomment-364199

**Reply by flauc** [reply]
Happy to help :)
